docs/jwst/emicorr/description.rst

```diff
@@ -6,65 +6,83 @@ Description
+#. Subtract the scaled slope image and bias from each frame of each integration.
 
-#. Use look-up table to get the aligned reference wave value for each pixel
-   (the noise array corresponding to the input image). [#f1]_
+#. Calculate the phase of every pixel in the image at the desired EMI frequency
+   (e.g. 390 Hz), relative to the first pixel in the image.
 
-#. Subtract the noise array from the input image and return the cleaned result.
+#. Make a binned phased amplitude (PA) waveform from the cleaned data.
 
-.. [#f1] Alternately, use the binned pa wave instead of the reference wave to "self-correct".
+#. Measure the phase shift between the binned PA waveform and the input
+   reference waveform. [#f1]_
 
-The long term plan is a change to the sizes and locations of the subarrays
-to get the frame times to be in phase with the known noise frequencies like
-the full frame images. For the previous and near term observations this can
-be fixed through application of the ``emicorr`` step.
+#. Use a look-up table to get the aligned reference waveform value for each pixel
+   (the noise array corresponding to the input image). [#f1]_
+
+#. Subtract the noise array from the input image and return the cleaned result.
 
-An EMICORR reference file can be used to correct for all known noise
-patterns. The reference file is expected to be in ASDF format, containing
-the exact frequency numbers, the corresponding 1D array for the phase
-amplitudes, and a ``subarray_cases`` dictionary that contains
-the frequencies to correct for according to subarray, read pattern, and
-detector. If there is no reference file found in CRDS, the step has a set
-of default frequencies and subarray cases for which the correction is
-applied.
+.. [#f1] Alternately, if a reference waveform is not available, use the binned
+   PA waveform to "self-correct" the noise.
+
+The joint algorithm proceeds similarly, except that the linear ramps and
+the EMI noise are fit simultaneously. It works by choosing pixels with modest
+scatter among the reads, and then finding the amplitude and phase of a supplied
+reference waveform that, when subtracted, makes these pixels' ramps as straight
+as possible. The straightness of the ramps is measured by a chi-squared metric, after
+fitting lines to each one.  As for the sequential algorithm, the noise at each
+frequency is fit and removed iteratively, for each desired frequency.
+
+The optimal choice for the fitting algorithm depends on the input.
+The sequential algorithm can be used without a reference waveform, generating
+a new reference file on-the-fly for user-specified frequencies, but it
+requires 10 or more groups for a reliable fit.  The joint algorithm
+requires a reference waveform but can successfully fit EMI in ramps with
+3 or more groups.  Currently, the default algorithm is "sequential", but users
+should consider specifying the joint algorithm if the number of groups
+in the input data is less than 10.
```

Comment:
I'd be inclined to remove the statement about which algorithm is the default from this part of the documentation.  The top-level 'algorithm' argument already gives the default, and I'd worry that this additional statement might get missed in any future update (particularly if changed via param ref file rather than code default).  Suggest just removing this last sentence.